### PR TITLE
watchdog timer

### PR DESCRIPTION
**Pull request by AdityaHarvi**
Watchdog timer to ensure that all players are still in the game.
This fixes situations where all players leave the game. If no playerMove event was received by the server for 5 seconds, then kill the game and restart.

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -123,6 +130,21 @@ const spectator = new Spectator(
 const scene = new SceneTracker();
 const boardSync = new BoardSync(updateBoard);
 
+let watchdogTimer: Watchdog;
+const watchdogFood = {
+    data: "yummy",
+    timeout: TIMEOUT,
+};
```
Comment (swebra) [inline server/index.ts:137]: is the yummy really needed here lol

Comment (AdityaHarvi) [reply server/index.ts:137]: I know, I know... it causes errors without *some* data property

Comment (AdityaHarvi) [reply server/index.ts:137]: the author of the library REALLY likes to simulate this to an actual dog

**Comment by swebra** [comment]
To be clear, the reset occurs if no player out of the four has communicated to the server in 5 seconds, right?

**Comment by AdityaHarvi** [comment]
> To be clear, the reset occurs if no player out of the four has communicated to the server in 5 seconds, right?
 Correct.
Upon receiving a playerMove event, the dog is fed (timer is reset).
Upon entering the gameOverScene, the dog goes to sleep (timer is cleared).
Upon entering gameArenaScene, a new dog is created

**Comment by swebra** [comment]
Are there any concerns about reconnecting clients that are out of sync with the scene and/or the player ids? Like the server blips out for a bit and resets back to waiting room. The previously connected clients are still in the game arena, and may reconnect and be sending arena messages to the server. Maybe that's out of scope for this PR (and the capstone deadline)

**Comment by AdityaHarvi** [comment]
I think thats a general problem with our game. Its definitely out of scope of this PR but I'm not sure if we event want to be attempting to fix something like that 🤔

**Review by swebra — APPROVED** [review]

**Pull request merged**